File: GamingBlogAPI/src/main/java/cz/osu/gamingblogapi/responses/PostResponse.java
package cz.osu.gamingblogapi.responses;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PostResponse {
    private long id;
    private String title;
    private String description;
    private String htmlContent;
    private String thumbnailUrl;
    private CategoryResponse category;
    private AuthorResponse author;
}

File: gaming-blog-react-web-client/src/Responses/PostResponse.ts
import {CategoryResponse} from "./CategoryResponse";
import {AuthorResponse} from "./AuthorResponse";

export interface PostResponse {
    id: number;
    title: string;
    description: string;
    htmlContent: string;
    thumbnailUrl: string;
    category: CategoryResponse;
    author: AuthorResponse;
}